src/thread_pool.rs

```diff
@@ -116,6 +139,18 @@ impl Registry {
         self.work_available.notify_all();
     }
 
+    pub unsafe fn inject_deferred(&self, job: Box<Job>) {
+        log!(InjectJobs { count: 1 });
+        let mut state = self.state.lock().unwrap();
+        if state.terminate {
+            drop(state);
+            // See the comment in inject about this case.
+            panic!("rayon thread pool is contaminated by a previous panic; recovery is only available on nightly compilers");
```

Comment:
`recovery is only available on nightly compilers` - is this talking about the `std` panic or Rayon's propagation of the panics?
If it is the std, it is available since Rust 1.9(Stable).